SimPEG/regularization/rotated.py

```diff
@@ -0,0 +1,241 @@
+            if mesh.dim == 2:
+                anis_alpha = np.stack(
+                    (
+                        anis_alpha[..., 0, 0],
+                        anis_alpha[..., 1, 1],
+                        anis_alpha[..., 0, 1],
+                    ),
+                    axis=-1,
+                )
+            elif mesh.dim == 3:
+                anis_alpha = np.stack(
+                    (
+                        anis_alpha[..., 0, 0],
+                        anis_alpha[..., 1, 1],
+                        anis_alpha[..., 2, 2],
+                        anis_alpha[..., 0, 1],
+                        anis_alpha[..., 0, 2],
+                        anis_alpha[..., 1, 2],
+                    ),
+                    axis=-1,
+                )
+        self._anis_alpha = anis_alpha
+
+    # overwrite the call, deriv, and deriv2...
+    def __call__(self, m):
+        G = self.cell_gradient
+        M_f = self.W
+        r = G @ (self.mapping * (self._delta_m(m)))
+        return 0.5 * r @ M_f @ r
+
+    def deriv(self, m):
+        m_d = self.mapping.deriv(self._delta_m(m))
+        G = self.cell_gradient
+        M_f = self.W
+        r = G @ (self.mapping * (self._delta_m(m)))
+        return m_d.T * (G.T @ (M_f @ r))
+
+    def deriv2(self, m, v=None):
+        m_d = self.mapping.deriv(self._delta_m(m))
+        G = self.cell_gradient
+        M_f = self.W
+        if v is None:
+            return m_d.T @ (G.T @ M_f @ G) @ m_d
+
+        return m_d.T @ (G.T @ (M_f @ (G @ (m_d @ v))))
+
+    @property
+    def cell_gradient(self):
+        """The (approximate) cell gradient operator
+
+        Returns
+        -------
+        scipy.sparse.csr_matrix
+        """
+        if getattr(self, "_cell_gradient", None) is None:
+            mesh = self.regularization_mesh.mesh
+            try:
+                cell_gradient = mesh.cell_gradient
+            except AttributeError:
+                a = mesh.face_areas
+                v = mesh.average_cell_to_face @ mesh.cell_volumes
+                cell_gradient = sp.diags(a / v) @ mesh.stencil_cell_gradient
+
+            v = np.ones(mesh.n_cells)
+            # Turn off cell_gradient at boundary faces
+            if self.active_cells is not None:
+                v[~self.active_cells] = 0
+
+            dv = cell_gradient @ v
+            P = sp.diags((np.abs(dv) <= 1e-16).astype(int))
+            cell_gradient = P @ cell_gradient
+            if self.active_cells is not None:
+                cell_gradient = cell_gradient[:, self.active_cells]
+            self._cell_gradient = cell_gradient
+        return self._cell_gradient
+
+    @property
+    def W(self):
+        """The inner product operator using rotated coordinates
+
+        Returns
+        -------
+        scipy.sparse.csr_matrix
+        """
+        if getattr(self, "_W", None) is None:
+            mesh = self.regularization_mesh.mesh
+            cell_weights = np.ones(len(mesh))
```

Comment:
@jcapriot : I don't think this works with `active_cells` because `len(mesh)` here will be the size of the total mesh (active + inactive), but later, you multiply those `cell_weights` by `values` (such as sensitivity weights) which will be the shape of the number of inactive cells.

See the error I am getting:
<img width="813" alt="image" src="https://github.com/simpeg/simpeg/assets/97514898/9141d88a-1b04-4476-adb4-9b617865a630">

Reply:
PS: the issue only appears if you are setting weights AND have active_cells.